src/components/sbb-form-field/sbb-form-field.tsx

```diff
@@ -0,0 +1,213 @@
+  }
+
+  /**
+   * It is used internally to assign the attributes of `<input>` to `_id` and `_input` and to observe the native readonly and disabled attributes.
+   */
+  private _onSlotInputChange(event: Event): void {
+    this._input = (event.target as HTMLSlotElement).assignedElements()[0] as HTMLElement;
+    this._originalInputAriaDescribedby = this._input.getAttribute('aria-describedby');
+
+    this._readonly = this._input.hasAttribute('readonly');
+    this._disabled = this._input.hasAttribute('disabled');
+
+    this._formFieldAttributeObserver.observe(this._input, {
+      attributes: true,
+      attributeFilter: ['readonly', 'disabled', 'class'],
+    });
+
+    if (!this._input.id) {
+      this._input.id = `sbb-form-field-input-${nextId++}`;
+    }
+  }
+
+  /**
+   * @private
+   * It is used internally to set the focus in the `<input>`.
+   */
+  private _setFocus(): void {
+    this._input.focus();
+  }
+
+  /**
+   * @param mutationsList The list of the attributes
+   * @private
+   * It is used internally to bind on the `MutationObserver`.
+   */
+  private _onAttributesChange(mutationsList: MutationRecord[]): void {
+    for (const mutationRecord of mutationsList) {
+      if (mutationRecord.type !== 'attributes') {
+        return;
+      }
+      const inputEl = mutationRecord.target as HTMLInputElement;
+      this._disabled = inputEl.hasAttribute('disabled');
+      this._readonly = inputEl.hasAttribute('readonly');
+      const list = inputEl.classList;
+      this._invalid =
+        list.contains('sbb-invalid') ||
+        (list.contains('ng-touched') && list.contains('ng-invalid'));
+    }
+  }
+
+  public render(): JSX.Element {
+    return (
+      <Host
+        class={{
+          [`form-field--error-space-${this.errorSpace}`]: true,
+          [`form-field--size-${this.size}`]: true,
+          'form-field--borderless': this.borderless,
+          'form-field--readonly': this._readonly,
+          'form-field--disabled': this._disabled,
+          'sbb-invalid': this._invalid,
+        }}
+      >
+        {/* eslint-disable-next-line jsx-a11y/click-events-have-key-events,jsx-a11y/no-static-element-interactions */}
+        <div onClick={(): void => this._setFocus()} class="form-field__wrapper">
+          <div
+            class={{
+              'form-field__prefix': true,
+              'form-field__prefix--empty': !this._prefixElements.length,
+            }}
+          >
+            <slot
+              name="prefix"
+              onSlotchange={(event): unknown =>
+                (this._prefixElements = (event.target as HTMLSlotElement).assignedElements())
+              }
+            ></slot>
+          </div>
+          <div class="form-field__input-container">
+            <label class="form-field__label" htmlFor={this._input?.id}>
+              <slot name="label">
+                <span>{this.label}</span>
+              </slot>
+              {this.optional && <span>&nbsp;(optional)</span>}
```

Comment:
"Optional" string has to be translated, see the old sbb-input component on how to achieve this.